File: api/services/temperature_db.rb
def retrieve_temperatures_from_db(location, start_date_str, end_date_str)
  temperatures_from_db = location.temperature_models.where(date: { '$gte' => start_date_str, '$lte' => end_date_str })
  list_temperatures = temperatures_from_db.map do |temperature|
    {
      'date' => temperature.date,
      'min-forecasted' => temperature.min_forecasted,
      'max-forecasted' => temperature.max_forecasted
    }
  end
  return list_temperatures
end

def save_temperatures_in_db(location, list_dates)
  list_dates.each do |temp_data|
    temperature = location.temperature_models.find_or_initialize_by(date: temp_data['date'])
    temperature.min_forecasted = temp_data['min-forecasted']
    temperature.max_forecasted = temp_data['max-forecasted']
    unless temperature.save
      status 400
      return {error: "Error saving record: #{temperature.errors.full_messages.join(', ')}"}.to_json
    end
  end
end

File: nodejs_fetch_module/src/services/temperatureDb.js
// Function to save temperatures in DB
async function saveTemperaturesInDB(location, forecastData, temperaturesCollection) {
    // Find the temperatures associated with the location
    const existingTemperatures = await temperaturesCollection.find({ location_models_id: location._id }).toArray();
  
    // Update or insert temperatures for each date
    for (const tempData of forecastData) {
      const existingTemperature = existingTemperatures.find(temp => temp.date === tempData.date);
  
      if (existingTemperature) {
        // Update existing temperature record
        await temperaturesCollection.updateOne(
          { _id: existingTemperature._id },
          { $set: { min_forecasted: tempData['min-forecasted'], max_forecasted: tempData['max-forecasted'] } }
        );
      } else {
        // Insert new temperature record
        await temperaturesCollection.insertOne({
          location_models_id: location._id,
          date: tempData.date,
          min_forecasted: tempData['min-forecasted'],
          max_forecasted: tempData['max-forecasted'],
          updated_at: new Date(),
          created_at: new Date(),
        });
      }
    }
  
    return { success: 'Documents saved successfully' };
  }
  
  module.exports = saveTemperaturesInDB;
  

module.exports = saveTemperaturesInDB